src/routes/admin_moderation.py

```diff
@@ -0,0 +1,161 @@
+
+cloudinary.config(
+        cloud_name=settings.cloudinary_name,
+        api_key=settings.cloudinary_api_key,
+        api_secret=settings.cloudinary_api_secret,
+        secure=True
+    )
+
+
+
+# response_model=List[UserStatistics]
+@router.get('/statistics')
+async def get_user_statistics(
+                            current_user: User = Depends(auth_service.get_current_user),
+                            db: Session = Depends(get_db)):
+    # raise ValueError("Not implemented")
+    is_admin(current_user)
+    
+    return await repository_admin_moderation.get_user_statistics(db)
+
+
+@router.post('/add-photo', response_model=PhotoResponse)
+async def create_photo(file: UploadFile = File(),
+                       description: str = Form(),
+                       user_id: int = Form(),
+                       current_user: User = Depends(auth_service.get_current_user),
+                       db: Session = Depends(get_db)):
+    # Check if the current user is an admin
+    is_admin(current_user)
+
+    # Create a clean file identifier
+    clean_description = description[:7].replace(" ", "")
+    clean_filename = file.filename.replace(".", "")
+    public_id = f'PhotoShare/{clean_description}{clean_filename}'
+
+    try:
+        # Upload the file
+        r = cloudinary.uploader.upload(file.file, public_id=public_id, overwrite=True)
+
+        # Generate the Cloudinary image URL
+        url = cloudinary.CloudinaryImage(public_id).build_url(crop='fill', version=r.get('version'))
+
+        return await repository_admin_moderation.create_photo(user_id, description, url, db)
+    except AuthorizationRequired as e:
+        print(f"Required authorization: {e}.")
+    except BadRequest:
+        print(f"Loading ERROR. Bad request or file type not supported.")
+    except Error as e:
+        print(f"Error during loading the file: {e}")
+
+
+@router.get('/get-photo/{photo_id}', response_model=PhotoResponse)
+async def read_photo(photo_id: int,
+                     current_user: User = Depends(auth_service.get_current_user),
+                     db: Session = Depends(get_db)):
+    # Check if the current user is an admin
+    is_admin(current_user)
+
+    photo = await repository_admin_moderation.read_photo(photo_id, db)
+    if photo is None:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Photo not found")
+    return photo
+
+
+@router.put('/update-photo/{photo_id}', response_model=PhotoResponse)
+async def update_photo(photo_id: int,
+                       file: UploadFile = File(),
+                       description: str = Form(),
+                       user_id: str = Form(),
+                       current_user: User = Depends(auth_service.get_current_user),
+                       db: Session = Depends(get_db)):
+    # Check if the current user is an admin
+    is_admin(current_user)
+
+    # Create a clean file identifier
+    clean_description = description[:7].replace(" ", "")
```

Comment:
clean_description = description[:7].replace(" ", "")
    clean_filename = file.filename.replace(".", "")
    public_id = f'PhotoShare/{clean_description}{clean_filename}'

    try:
        # Upload the file
        r = cloudinary.uploader.upload(file.file, public_id=public_id, overwrite=True)

        # Generate the Cloudinary image URL
        url = cloudinary.CloudinaryImage(public_id).build_url(crop='fill', version=r.get('version'))

the redundancy - can be moved out into the function